cou/cli.py

```diff
@@ -42,110 +71,333 @@ def parse_args(args: Any) -> argparse.Namespace:
+    # Configure top level argparser and its options
     parser = argparse.ArgumentParser(
-        description="Charmed OpenStack Upgrader(cou) is an application to upgrade Charmed "
-        "OpenStack. Application identifies the lowest OpenStack version on the components and "
-        "upgrade to the next version.",
-        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="Canonical OpenStack Upgrader(cou) is an application to upgrade Canonical "
+        "OpenStack cloud. Application identifies the lowest OpenStack version on the components "
+        "and upgrade to the next version.",
+        formatter_class=CapitalisedHelpFormatter,
+        usage="%(prog)s [options] <command>",
         exit_on_error=False,
         add_help=False,
+        allow_abbrev=False,
     )
     parser.add_argument(
-        "--run",
-        help="Use this flag to run the upgrade, otherwise just print out the upgrade steps.",
-        action="store_true",
-        default=False,
+        "--version",
+        "-V",
+        action="version",
+        default=argparse.SUPPRESS,
+        help="Show version details.",
+        version=pkg_resources.require("canonical_openstack_upgrader")[0].version,
     )
     parser.add_argument(
-        "--log-level",
-        default="INFO",
-        dest="loglevel",
-        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
-        type=str.upper,
-        help="Set the logging level. Defaults to INFO. This only affects stdout. The logfile "
-        "will be always DEBUG. The file location is in COU_DATA/log. You can set the "
-        "COU_DATA via environment variable. However it needs to be plugged to snap.",
+        "--help",
+        "-h",
+        action="help",
+        default=argparse.SUPPRESS,
+        help="Show this help message and exit.",
     )
-    parser.add_argument(
+
+    # Configure subcommand parser
+    subparsers = parser.add_subparsers(
+        title="Commands",
+        dest="command",
+        help="For more information about a command, run 'cou help <command>'.",
+    )
+
+    # Arg parser for "cou help" sub-command
+    help_parser = subparsers.add_parser(
+        "help",
+        usage="cou help [command]",
+    )
+    help_parser.add_argument(
+        "subcommand",
+        nargs="?",
+        choices=["plan", "run", "all"],
+        default="all",
+        type=str,
+        help="A sub-command to get information of.",
+    )
+
+    # Define common options for subcommands and their child commands
+    subcommand_common_opts_parser = argparse.ArgumentParser(add_help=False)
+    subcommand_common_opts_parser.add_argument(
         "--model",
         default=None,
         dest="model_name",
+        type=str,
         help="Set the model to operate on. If not set it gets the model name in this order:\n"
         "  1 - Environment variable JUJU_MODEL,"
         "  2 - Environment variable MODEL_NAME,"
         "  3 - Current active juju model",
```

Comment:
the newlines seem to be ignored on my terminal:
```
$ canonical-openstack-upgrader.cou plan --help
Usage: cou plan [options]

Show the steps for upgrading the cloud to the next release.

options:
  -h, --help            show this help message and exit
  --model MODEL_NAME    Set the model to operate on. If not set it gets the model name in this order: 1 - Environment variable JUJU_MODEL, 2 - Environment variable MODEL_NAME, 3 - Current active juju model
  --parallel            Run upgrade steps in parallel where possible.
  --verbose, -v         Increase logging verbosity in STDOUT. Repeat the 'v' in the short option for more detail. Maximum verbosity is obtained with 4 (or more) v's, i.e. -vvvv. Note that this doesn't affect the
                        verbosity in logfile, which will always have the maximum verbosity.
  --quiet, -q           Disable output in STDOUT.

Upgrade group:
  {control-plane,data-plane}
                        For more information about a upgrade group, run 'cou plan <upgrade-group>' -h.
    control-plane       Show the steps for upgrading the control-plane components.
    data-plane          Show the steps for upgrading the data-plane components.
```

Reply:
Update. The newlines are displayed in the intended format now